docs/source/developer_documentation.rst

```diff
@@ -26,20 +26,22 @@ A variety of coding conventions are enforced by automated tools in continuous in
 PyRenew Principles
 ------------------
 
--  Variable naming conventions
+Variable naming conventions
+---------------------------
+-  Use the ``data_`` prefix for (potentially) observed data.
+-  Use the ``_rv`` suffix for random variables.
+-  Use the ``observed_`` for the output of sample statements where ``obs`` is a ``data_`` prefixed object.
+-  Thus, code which may reasonably written like ``infections = infections.sample(x, obs=infections)`` should instead be written ``observed_infections = infections_rv.sample(x, obs=data_infections)``.
 
-   -  Use the ``data_`` prefix for (potentially) observed data.
-   -  Use the ``_rv`` suffix for random variables.
-   -  Use the ``observed_`` for the output of sample statements where ``obs`` is a ``data_`` prefixed object.
-   -  Thus, code which may reasonably written like ``infections = infections.sample(x, obs=infections)`` should instead be written ``observed_infections = infections_rv.sample(x, obs=data_infections)``.
 
--  Class conventions
+Class conventions
+-----------------
+-  Composing ``Model`` objects is discouraged. To modularize and reuse code across ``Model``s, create custom ``RandomVariable`` classes.
+-  Returning anything from ``Model.sample`` is discouraged. Instead, sample from models using ``Predictive`` or our ``prior_predictive`` or ``posterior_predictive`` functions.
+-  Using ``numpyro.deterministic`` within a ``RandomVariable.sample`` method is discouraged. Only use it in ``Model.model`` calls. If something might need to be recorded from a ``RandomVariable``, it should be returned from the ``RandomVariable.sample`` call so it can be recorded within the ``Model.model`` call.
+-  Using default site names in a ``RandomVariable.sample`` is discouraged. Only use default site names at the ``Model.model`` level.
```

Comment:
I think there is a better term than “using.” This is meant to suggest that someone writing a random variable should not give their own random variable a default site name. We should say something closer to that and point out that the site name should be an argument to the RV constructor.

Reply:
Tried something in 33063a7. Might be too verbose. Let me know what you think, @damonbayer